openmls/src/group/mls_group/builder.rs

```diff
@@ -198,4 +199,12 @@ impl MlsGroupBuilder {
             .external_senders(external_senders);
         self
     }
+
+    /// Sets the initial group context extensions
+    pub fn with_group_context_extensions(mut self, extensions: Extensions) -> Self {
```

Comment:
Do we check that `extensions` are only valid group context extensions?
This is of course a little difficult for extensions that OpenMLS doesn't know about, and the spec doesn't give us much to help here. But we should at least make sure that we don't allow extensions in here from the RFC that are not group context extensions. And we should have a test case for that.